front/pages/api/lookup.ts

```diff
@@ -0,0 +1,192 @@
+import { getSession as getAuth0Session } from "@auth0/nextjs-auth0";
+import type { WithAPIErrorResponse } from "@dust-tt/types";
+import { isLeft } from "fp-ts/Either";
+import * as t from "io-ts";
+import * as reporter from "io-ts-reporters";
+import type { NextApiRequest, NextApiResponse } from "next";
+
+import config from "@app/lib/api/config";
+import { getBearerToken } from "@app/lib/auth";
+import { getPendingMembershipInvitationWithWorkspaceForEmail } from "@app/lib/iam/invitations";
+import type { SessionWithUser } from "@app/lib/iam/provider";
+import { fetchUserFromSession } from "@app/lib/iam/users";
+import { findWorkspaceWithVerifiedDomain } from "@app/lib/iam/workspaces";
+import { Workspace } from "@app/lib/models/workspace";
+import { apiError, withLogging } from "@app/logger/withlogging";
+
+export type BaseLookupResponse = {
+  isNew: boolean;
+};
+
+export type UserLookupResponse = BaseLookupResponse & {
+  hasInvite: boolean;
+  hasAutoJoinWorkspace: boolean;
+  workspaceId?: string;
+};
+
+export type WorkspaceLookupResponse = BaseLookupResponse;
```

Comment:
why all the exports ?

Reply:
The typing feels too complicated, I would have 2 distinct types, it's just simpler.